lib/internal/modules/esm/module_job.js

```diff
@@ -75,6 +75,19 @@ const isCommonJSGlobalLikeNotDefinedError = (errorMessage) =>
 const explainCommonJSGlobalLikeNotDefinedError = (e, url) => {
   if (e?.name === 'ReferenceError' &&
       isCommonJSGlobalLikeNotDefinedError(e.message)) {
+
+    if (e.message.includes('require is not defined')) {
+      e.message = [
+        'ERR_AMBIGUOUS_MODULE_SYNTAX: This file cannot be parsed as either CommonJS or ES Module.',
+        '- CommonJS error: await is only valid in async functions.',
+        '- ES Module error: require is not defined in ES module scope.',
+        'If you meant to use CommonJS, wrap top-level await in async function.',
+        'If you meant to use ESM, do not use require().',
+      ].join('\n');
```

Comment:
Creating an array of strings and joining them is unnecessary processing to do at runtime. It would be more performant to just define a single string in the first place.

Also I can't think of other instances where we return multiline error messages. Unless this is a pattern we already use elsewhere, this should be rephrased to be one line. It doesn't need to be a single sentence, but it shouldn't include a newline.

Reply:
I absolutely agree, I have updated this as a single line